pkg/operator/starter.go

```diff
@@ -228,3 +234,59 @@ func (v *versionGetter) VersionChangedChannel() <-chan struct{} {
 	// this versionGetter never notifies of a version change, getVersion always returns the new version.
 	return make(chan struct{})
 }
+
+// ensureDaemonSetCleanup continually ensures the removal of the daemonset
+// used to manage controller-manager pods in releases prior to 4.12. The daemonset is
+// removed once the deployment now managing controller-manager pods reports at least
+// one pod available.
+func ensureDaemonSetCleanup(ctx context.Context, kubeClient *kubernetes.Clientset, eventRecorder events.Recorder) {
+	// daemonset and deployment both use the same name
+	resourceName := "controller-manager"
+
+	dsClient := kubeClient.AppsV1().DaemonSets(util.TargetNamespace)
+	deployClient := kubeClient.AppsV1().Deployments(util.TargetNamespace)
+
+	go wait.UntilWithContext(ctx, func(_ context.Context) {
+		// This function isn't expected to take long enough to suggest
+		// checking that the context is done. The wait method will do that
+		// checking.
+
+		// Check whether the legacy daemonset exists and is not marked for deletion
+		ds, err := dsClient.Get(ctx, resourceName, metav1.GetOptions{})
```

Comment:
This will generate on Get request every minute. Any way to completely exit the loop when the DS it not found?

Reply:
+1, exiting the loop now - the DS is only managed by the operator and should not be recreated again